src/mo_interpreter/interpret.ml

```diff
@@ -620,10 +620,17 @@ and interpret_exp_mut env exp (k : V.value V.cont) =
     interpret_exp env exp1 (fun v1 ->
       interpret_cases env cases exp.at v1 k
       )
-  | TryE (exp1, cases) ->
+  | TryE (exp1, cases, None) ->
     let k' = fun v1 -> interpret_catches env cases exp.at v1 k in
     let env' = { env with throws = Some k' } in
     interpret_exp env' exp1 k
+  | TryE (exp1, cases, Some exp2) ->
+    let k' v1 =
+      let cleanup v2 = interpret_exp env exp2 (fun _ -> k v2) in
+      interpret_catches env cases exp.at v1 cleanup in
+    let env' = { env with throws = Some k' } in
```

Comment:
I'm not sure this is enough. Unless we disallow returns and break from exp1 and exp2, I think you will need to redefine the rets ands labs continuations too, execute the finally before proceeding. This is getting complicated....

Reply:
I think your current code will allow a `return` (or `break`) from `exp1` or `cases` to skip the finally clause - only if the execution falls through `exp1` or `cases` is the finally clause executed. A throw from `cases` will probably also bypass the finally.

Reply:
Probably worth having a careful read of this https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/statements#1311-the-try-statement

Reply:
Right. We now enter the `try` body with all possible continuations (success, `throw`/failure, `return` and `label`) augmented. I also added tests for all these cases.